data_source_examples/slack_conversation_history/using_airtable.js/README.md

```diff
@@ -0,0 +1,66 @@
+- If you choose other field names, be sure to update the code in [index.js](./index.js) in the `convertSlackMessageToAirtableRecord` function.
+
+### B. Slack App Setup
+
+Follow these instructions to set up a Slack app, retrieve an API token, and add your new bot to the channel(s) you want to extract history from.
+
+1. Create a custom app from https://api.slack.com/apps?new_app=1, making sure to select the Slack workspace that contains the channel you want to extract messages from.
+2. Add a **bot token scope** from your new app's 'OAuth & Permissions' page. If you are extracting messages from a public mesage, you'll want to add `channels:history`. For private channels add `groups:history`. For DMs or MPDMs, use `im:history` or `mpim:history`, respectively. 
+3. Retrieve a **bot user token** for your workspace by clicking 'Install to Workspace' near the top of your app's 'OAuth & Permissions' page. This may require Slack admin approval.
+4. Once the app has been installed, be sure to add the app to the channel you want to extract messages from. You can type `/invite @<your app name here>`.
+
+### C. Configuring and Running this Script
+
+Finally, let's setup this script to run locally. You can later deploy this as a scheduled task on a server or cloud function.
+
+1. Clone/unzip code
+2. Copy `.env.example` to `.env` and populate values (see below for details on each environment variable)
+3. Install node dependencies including Airtable and Slack's official SDKs by running `npm install`
+4. Trigger the script to execute by running `npm run sync`
+5. Records in the specified Airtable base's table should be created. Try adding new messages to the Slack channel and re-run the previous step.
+ 
+### Key Files and Their Contents
+
+- [`index.js`](index.js) is the main code file which is executed when `npm start` is run. At a high level, it performs the following:
+  - Loads dependencies, helper functions, and configuration variables
+  - Initializes API clients for the Airtable and Slack APIs
+  - Retrieves Slack messages (both "parent" and "threaded replies") from the Slack API and transforms the objects to be flat and suitable for our base template (setup section A)
+  - Retrieves all existing records in the Airtable base and creates a mapping of the unqiue field's value to the existing record ID for later updating
+  - First loops through each of the parent messages, determining if a new record needs to be created or an existing record updated. This step then repeats for all threaded replies.
+    - Note: This example is more complicated than the generic examples as we first loop through all parent Slack messages to update-or-create them and then loop through all threaded replies so that they can be properly associated them with the correct parent Slack message. This can be observed in [index.js](./index.js); look for `// First, upsert all parent messages, and then upsert all threaded replies`
```

Comment:
If you want to be extra I think you can link to a line number

Reply:
100% but I want to try to protect against breaking that link in the future